src/core/viz/expressions/Animation.js

```diff
@@ -93,7 +96,46 @@ export class Animation extends BaseExpression {
         return !this.paused;
     }
 
+    _dataReady() {
+        if (waitingForLayer.has(this)) {
+            waitingForLayer.delete(this);
+            waitingForOthers.add(this);
+        }
+        if (waitingForOthers.has(this)) {
+            waitingForLayer = new Set([...waitingForLayer].filter(expr=>{
+                while (expr.parent){
+                    expr = expr.parent;
+                }
+                if (expr._getRootExpressions){
+                    // The animation hasn't been removed from the viz
+                    return true;
+                }
+                return false;
+            }));
+            if (waitingForLayer.size > 0) {
+                return;
+            }
+            [...waitingForOthers.values()].map(anim => {
+                if (anim._paused === 'default') {
+                    anim._paused = false;
+                }
+            });
+            waitingForOthers.clear();
+        }
+    }
+
+    _postShaderCompile(program, gl) {
+        waitingForLayer.add(this);
+        this._paused = 'default';
+        super._postShaderCompile(program, gl);
+    }
+
     _setTimestamp(timestamp) {
+        super._setTimestamp(timestamp);
+
+        if (this._paused && this._lastTime === undefined) {
```

Comment:
This should warrant some comment for the future... although I understand anything you write could be potentially more puzzling